### PR TITLE
Android: documentation updates for SDK 10.1

### PR DESCRIPTION
**Pull request by makohoek**
This is a major documentation update, to prepare for the 10.1 SDK which will include:
- Android 15 support
- New bootloader environment for compiling everything easily (using `repo`)
- Android15-6.6 kernel
- Bootconfig and bootmeth support in bootloader

Please see each individual commit for details.

cc: @vishalmti @glaroque

Thanks
Mattijs

### CODE REVIEW

**Comment by makohoek** [comment]
`rstcheck` is wrong here:

```
λ /mnt/work/ti/src/processor-sdk-doc/ mkorpershoek/android-sdk-10.1 rg 'android-ota-sideload'
source/devices/AM62PX/android/Application_Notes_OTA.rst
9:1. :ref:`android-ota-sideload`
13:.. _android-ota-sideload:

source/devices/AM62X/android/Application_Notes_OTA.rst
9:1. :ref:`android-ota-sideload`
13:.. _android-ota-sideload:
```

We can see that the hyperlink is indeed referenced in both files.

File: source/devices/AM62X/android/Release_Specific_Release_Notes.rst [hunk 1]
```diff
@@ -57,17 +57,17 @@ SDK Components and Versions
...
 +------------------------------------+-------------------------------------------------------------------------------+
-| **Linux Kernel**                   | Linux 6.1.78 (6.6.30 preview available)                                       |
+| **Linux Kernel**                   | Linux 6.6.56                                                                  |
 +------------------------------------+-------------------------------------------------------------------------------+
 | **Linux Kernel Toolchain**         | clang-r487747c                                                                |
 +------------------------------------+-------------------------------------------------------------------------------+
-| **U-boot**                         | 2023.04 (2024.04 preview available)                                           |
+| **U-boot**                         | 2024.04                                                                       |
 +------------------------------------+-------------------------------------------------------------------------------+
 | **GCC Toolchain**                  | GNU Toolchain for the A-profile Architecture 13.2 (rel1)                      |
```
Comment (cshilwant) [inline source/devices/AM62X/android/Release_Specific_Release_Notes.rst:68]: @makohoek 
Just curious, 

Is there a particular for this to be 13.2 & not 13.3 similar to Yocto - https://software-dl.ti.com/processor-sdk-linux/esd/AM62X/10_00_07_04/exports/docs/devices/AM62X/linux/Release_Specific_Release_Notes.html#what-s-new

**Review by cshilwant — CHANGES_REQUESTED** [review]
https://github.com/TexasInstruments/processor-sdk-doc/pull/68#discussion_r1869413146

File: source/devices/AM62X/android/Release_Specific_Release_Notes.rst (hunk 1 above)
Comment (makohoek) [reply source/devices/AM62X/android/Release_Specific_Release_Notes.rst]: Fixed in v2.

(In parallel, i'm working on patching https://git.ti.com/cgit/android/build-bootloaders/tree/utils.sh?h=d-android15-release#n51 but I've confirmed that it builds fine already)

**Comment by cshilwant** [comment]
https://github.com/TexasInstruments/processor-sdk-doc/pull/68#issuecomment-2517313361

@makohoek Seems like you need to have many references of code block directive without a specified language.

**Comment by makohoek** [comment]
> @makohoek Seems like you need to have many references of code block directive without a specified language.

I see only 4 that are android related in https://github.com/TexasInstruments/processor-sdk-doc/pull/68#issuecomment-2517313361

I did not know it was mandatory to add a language for each block. I will do it for next version. thanks for the review!

**Comment by cshilwant** [comment]
> I did not know it was mandatory to add a language for each block. I will do it for next version. thanks for the review!

Yeah, it just got added few hours back after we merged - https://github.com/TexasInstruments/processor-sdk-doc/pull/61 :wink: 
Specifically https://github.com/TexasInstruments/processor-sdk-doc/commit/2a6b81efea2bebb3b068e46a815741bc8e2e40a0 which made rstcheck more exhaustive

**Comment by makohoek** [comment]
Change in v3:
- Add language to each code-block listing

**Comment by makohoek** [comment]
The errors are still listed for Android but there is no code-block without language anymore:

```
rg 'code-block::$' android devices/AM62X/android devices/AM62PX/android
# empty 
```

**Comment by makohoek** [comment]
Change in v4: rebased

**Review by StaticRocket — CHANGES_REQUESTED** [review]

File: source/devices/AM62PX/android/Application_Notes_OTA.rst [hunk 2]
```diff
@@ -16,21 +16,21 @@ There are 2 ways of applying an OTA update:
 OTA via ``adb sideload``
 ************************
 
-1. Build and flash android images following :ref:`android-build-aosp` instructions.
+#. Build and flash android images following :ref:`android-build-aosp` instructions.
```
[the remark this answers is not in the document]
Comment (makohoek) [reply source/devices/AM62PX/android/Application_Notes_OTA.rst:19]: Right, thanks a lot for catching this. I've made a dedicated commit to fix this in v5.

File: source/devices/AM62X/android/Application_Notes_OTA.rst [hunk 3]
```diff
@@ -16,21 +16,21 @@ There are 2 ways of applying an OTA update:
...
+#. Build OTA package
 
 .. code-block:: console
 
-   $ cd ${YOUR_PATH}/ti-aosp-14
+   $ cd ${YOUR_PATH}/ti-aosp-15
    $ source build/envsetup.sh
    $ lunch <BUILD_TARGET>
    $ export DIST_DIR=./dist_output
    $ m dist
```
[the remark this answers is not in the document]
Comment (makohoek) [reply source/devices/AM62X/android/Application_Notes_OTA.rst]: Right, thanks a lot for catching this. I've made a dedicated commit to fix this in v5.

File: source/devices/AM62PX/android/Application_Notes_OTA.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (makohoek) [reply source/devices/AM62PX/android/Application_Notes_OTA.rst:19]: Fixed with: https://github.com/TexasInstruments/processor-sdk-doc/pull/68/commits/6f655429c0bfab3642322793b078966910284b77

File: source/devices/AM62X/android/Application_Notes_OTA.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (makohoek) [reply source/devices/AM62X/android/Application_Notes_OTA.rst]: Fixed with: https://github.com/TexasInstruments/processor-sdk-doc/pull/68/commits/6f655429c0bfab3642322793b078966910284b77

**Comment by makohoek** [comment]
Changes in v5:
- Fixed indentation in OTA guides (See: https://github.com/TexasInstruments/processor-sdk-doc/pull/68/commits/6f655429c0bfab3642322793b078966910284b77)

**Comment by makohoek** [comment]
Changes in v6:
- Fixed indentation for anonymous lists in OTA guide (thanks Randolph)

File: source/devices/AM62PX/android/Application_Notes_OTA.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (makohoek) [reply source/devices/AM62PX/android/Application_Notes_OTA.rst]: Should now be fixed with this commit https://github.com/TexasInstruments/processor-sdk-doc/pull/68/commits/ca0d759a08deae88d380fe441f5f2e5a837c8171

**Review by StaticRocket — APPROVED** [review]

**Comment by makohoek** [comment]
@cshilwant it seems that github thinks there are still requested changes (by you) on this PR. I think I've addressed everything. Am I missing something? 

Thanks again to both you and Randolph for all the review, I've learned a lot about sphinx thanks to both of you!

**Comment by StaticRocket** [comment]
@vishalmti

**Comment by cshilwant** [comment]
> @cshilwant it seems that github thinks there are still requested changes (by you) on this PR. I think I've addressed everything. Am I missing something?
> 
> Thanks again to both you and Randolph for all the review, I've learned a lot about sphinx thanks to both of you!

I believe the warnings which are thrown now can be ignored.
You have my +1.
Will wait for an ack from @vishalmti before merging

**Review by cshilwant — APPROVED** [review]

**Review by praneethbajjuri — APPROVED** [review]

**Pull request merged**